File: oshi-front/src/apis/response/goods/GetGoodsList.Response.Dto.ts
import GoodsResponseDto from "./Goods.Response.Dto";

export default interface GetGoodsListResponseDto {
    goodsList: GoodsResponseDto[]; // 여러 굿즈 정보를 담는 리스트
    totalPages: number;
    totalCount: number;
    currentPage: number;
}

File: ohsi-back/src/main/java/com/oshi/ohsi_back/dto/response/goods/GetGoodsListResponseDto.java
package com.oshi.ohsi_back.dto.response.goods;

import java.util.List;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class GetGoodsListResponseDto {
    private List<GoodsResponseDto> goodsList;
    private int totalPages;
    private int totalCount;

    @Builder
    public GetGoodsListResponseDto(List<GoodsResponseDto> goodsEntities, int totalPages, int totalCount) {
        this.goodsList = goodsEntities;
        this.totalPages = totalPages;
        this.totalCount = totalCount;
    }
}
